src/gr/shared.h

```diff
@@ -152,6 +152,27 @@ class Point3D {
 //        Scalar overlap_estimation = 0.2;
 //    };
 
+// TODO: Document better
+// Wrapper around (any) PointType to allow mutation of position of PointType
+// through copying the internal position.
+template<typename PointType>
+struct PosMutablePoint : public PointType
```

Comment:
Hey,
Small question on this class: why not keeping it as internal class of `MatchBase` ? As it is used only internally to properly handle `Eigen::Map` with const access, we shouldn't need it outside of `MatchBase` and child classes, right ?

Reply:
Indeed.

I populate PosMutablePoint in point filters as in [PointPairFilter.h#L58](https://github.com/necipfazil/OpenGR/blob/c995fffdb79a564455d046b9b0660df0bc16ba2a/src/gr/algorithms/PointPairFilter.h#L58). However, point filter does not need to know it is dealing with PosMutablePoint instances either. I will refactor both (move PosMutablePoint to MatchBase, abstract point type in point filter without mentioning PosMutablePoint) accordingly.

Reply:
solved https://github.com/STORM-IRIT/OpenGR/pull/31/commits/8fc03e9152df56cea0d507d7be5e16e22fee9149

PosMutablePoint is now a public member type of MatchBase class, propogated only to its children classes. Other classes, such as point filters, traits and functors are not explicity aware of this middle point type: they take it as template parameter.

The reasons why we can't make PosMutablePoint a protected member of MatchBase are the methods [MatchBase::getFirstSampled()](https://github.com/necipfazil/OpenGR/blob/8fc03e9152df56cea0d507d7be5e16e22fee9149/src/gr/algorithms/matchBase.h#L142) and [MatchBase::getSecondSampled()](https://github.com/necipfazil/OpenGR/blob/8fc03e9152df56cea0d507d7be5e16e22fee9149/src/gr/algorithms/matchBase.h#L147) (notice return type).

Reply:
Ok, I am fine with keeping `PosMutablePoint` public, as long as it is well documented.
It must be identified as a convenience class.